### PR TITLE
psh: ensure that deinit is always executed

### PR DESCRIPTION
**Pull request by mateusz-bloch**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by damianloew — COMMENTED** [review]
In my opinion this solution is reasonable. Please always remember that we provide a jira id 
in commit description and in case of fixing an issue `Fixes: ...` prefix

File: trunner/harness/psh.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ def __call__(self, result: TestResult) -> TestResult:
 
         result.set_stage(TestStage.RUN)
         test_result = self.next_harness(result)
+
```
Comment (nalajcie) [inline trunner/harness/psh.py:101]: unrelated change

**Review by damianloew — APPROVED** [review]
